env_store.go

```diff
@@ -77,11 +83,25 @@ func (s *EnvStore) GetProviderPaths() map[string][]string {
 // LoadProviderSecrets creates a new provider for each detected provider using a specified config.
 // It then asynchronously loads secrets using each provider and it's corresponding paths.
 // The secrets from each provider are then placed into a map with the provider name as the key.
-func (s *EnvStore) LoadProviderSecrets(providerPaths map[string][]string) (map[string][]provider.Secret, error) {
+func (s *EnvStore) LoadProviderSecrets(providerPaths map[string][]string, appConfig *common.Config) (map[string][]provider.Secret, error) {
```

Comment:
lets add appConfig under envstore property. this way we only need to pass it in the envstore constructor and not have to pass it internally between methods